extensions/amp-date-display/1.0/date-display.js

```diff
@@ -65,26 +86,52 @@ let EnhancedVariablesV2Def;
  * @param {!DateDisplayDef.Props} props
  * @return {PreactDef.Renderable}
  */
-export function DateDisplay(props) {
-  const {render, children} = props;
-  const data = /** @type {!JsonObject} */ (getDataForTemplate(props));
+export function DateDisplay({
+  datetime,
+  timestampMs,
+  timestampSeconds,
+  displayIn,
+  locale,
+  offsetSeconds,
+  render,
+  children,
+  ...rest
+}) {
+  const epoch = getEpoch(datetime, timestampMs, timestampSeconds);
+  const data = useMemo(
+    () => getDataForTemplate(epoch, displayIn, locale, offsetSeconds),
+    [epoch, displayIn, locale, offsetSeconds]
+  );
+
+  const rendered = useRenderer(render || DEFAULT_RENDER, data);
+  const isHtml =
+    rendered && typeof rendered == 'object' && '__html' in rendered;
+
   useResourcesNotify();
 
-  return render(data, children);
+  return (
+    <Wrapper
+      {...rest}
+      as="time"
+      datetime={data['iso']}
+      dangerouslySetInnerHTML={isHtml ? rendered : null}
+    >
+      {isHtml ? null : rendered || children}
+    </Wrapper>
+  );
 }
 
 /**
- * @param {!DateDisplayDef.Props} props
+ * @param {number|undefined} epoch
+ * @param {string|undefined} displayIn
+ * @param {string|undefined} locale
+ * @param {number|undefined} offsetSeconds
  * @return {!EnhancedVariablesV2Def}
  */
-function getDataForTemplate(props) {
-  const {
-    displayIn = '',
-    locale = DEFAULT_LOCALE,
-    offsetSeconds = DEFAULT_OFFSET_SECONDS,
-  } = props;
-
-  const epoch = getEpoch(props);
+function getDataForTemplate(epoch, displayIn, locale, offsetSeconds) {
+  displayIn = displayIn || DEFAULT_DISPLAY_IN;
+  locale = locale || DEFAULT_LOCALE;
+  offsetSeconds = offsetSeconds || DEFAULT_OFFSET_SECONDS;
```

Comment:
This won't work because we always call `getDataForTemplate()` with all args. The annoyance here is due to the #29293. As soon as we figure it out, we would need to do a pass over the the default assignment.

Reply:
I redid this and I think it looks a little better. I tagged #29293 to move defaults to props destructuring later. LMK WYT.